### ISSUE TITLE
all-the-icons font is not installed

### ISSUE THREAD

**Issue by koiuo** [nix-community/nix-doom-emacs#362]
I installed the emacs-doom as flake using home-manager

```
{
  inputs = {
    ...
    nix-doom-emacs.url = "github:nix-community/nix-doom-emacs";
  };

  outputs = { home-manager, nixpkgs, nix-doom-emacs, ... }@inputs: {
    nixosConfigurations = {
      ...
    };
    homeConfigurations =
      let
        ...
      in {
        koiuo = home-manager.lib.homeManagerConfiguration {
          pkgs = nixpkgs.legacyPackages."x86_64-linux";
          modules = [
            nix-doom-emacs.hmModule
            ./users/koiuo/home.nix
            ...
          ];
        };
      };
  };
}
```

I used `test/doom.d` as my private config.

Upon startup emacs won't display any icons.

Doing `M-x all-the-icons-install-fonts` fixes the issue.

Seems like this package should be installed by the module.

**Comment by ckiee** [comment]
It installs it for the whole user. NixOS provides [`fonts.fonts`](https://search.nixos.org/options?channel=unstable&show=fonts.fonts&from=0&size=50&sort=relevance&type=packages&query=fonts.fonts) and a [`emacs-all-the-icons-fonts`](https://search.nixos.org/packages?channel=unstable&show=emacs-all-the-icons-fonts&from=0&size=50&sort=relevance&type=packages&query=emacs-all-the-icons-fonts). 

It'd be neat if fontconfig had a environment variable for extra font paths and `all-the-icons.el` knew about it but otherwise this is not feasible from our environment. (excluding the *optional* HM module)

Maybe upstream is willing?

**Comment by koiuo** [comment]
@ckiee , I'd highly appreciate an explanation why icons were broken after the installation then, and what would be the correct way to fix the issue on NixOS.

It's my first time installing emacs, and it doesn't help, that I do this on NixOS (here the procedure is quite different from other OSes).

Thanks in advance

**Comment by ckiee** [comment]
@koiuo Let's both try `nix run github:nix-community/nix-doom-emacs` and see if it's consistent.

On my system (with a ~3week old nixos-unstable) I have:
```
ckie@cookiemonster ~ -> fc-list | rg 'all-the-' | head -n1
/nix/store/0r2jnr8nn2gzag3ga7ww737h7iw5wf6z-emacs-all-the-icons-fonts-5.0.0/share/fonts/all-the-icons/weathericons.ttf: Weather Icons:style=Regular
ckie@cookiemonster ~ -> fc-list | rg 'all-the-' | rg home
ckie@cookiemonster ~ ->
```
Same in Doom: (`describe-font`)
```
name (opened by): -unknown-all-the-icons-normal-normal-normal-*-16-*-*-*-*-0-iso10646-1
       full name: all-the-icons:pixelsize=16:foundry=unknown:weight=normal:slant=normal:width=normal:scalable=true
       file name: /nix/store/0r2jnr8nn2gzag3ga7ww737h7iw5wf6z-emacs-all-the-icons-fonts-5.0.0/share/fonts/all-the-icons/all-the-icons.ttf
```

See the [relevant section of my NixOS config](https://github.com/ckiee/nixfiles/blob/1163eb2a23c121422f36d43f2cc1d8eeeca0bc49/modules/fonts.nix#L26).

**Comment by koiuo** [comment]
Ok, I think I see now what was the issue: I had my emacs daemon running, perhaps (?) since the time I tried vanilla emacs (I suppose the doom-emacs hmModule won't kill an already running daemon).

Anyway, I removed the fonts installed by the emacs, update the font cache, restarted the deamon, and now all the outputs on all the steps look similar to yours, and icons are displayed almost fine (they aren't loaded at once, only after some interaction with emacs. But that doesn't seem to be a concern of this module).

Thanks for your time!
